### PR TITLE
remove enhanced lambda redundancies

### PR DESCRIPTION
**Pull request by jcstorms1**
### What does this PR do?

Removes redundancies in the enhanced lambda code

### Types of changes

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [x] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [x] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [ ] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog
- [ ] This PR passes the integration tests (ask a Datadog member to run the tests)
- [ ] This PR passes the unit tests

### CODE REVIEW

File: aws/logs_monitoring/enhanced_lambda_metrics.py [hunk 1]
```diff
@@ -451,24 +452,11 @@ def parse_metrics_from_report_log(report_log_line):
...
-            metric_point_value,
-        )
-
-        initial_duration.add_tags(tags)
-
-        metrics.append(initial_duration)
-
     for metric_name in METRICS_TO_PARSE_FROM_REPORT:
+        # check if init duration is present
+        if not regex_match.gropu(metric_name):
```
Comment (tianchu) [inline aws/logs_monitoring/enhanced_lambda_metrics.py]: "gropu" seems to be a typo

File: aws/logs_monitoring/enhanced_lambda_metrics.py [hunk 2]
```diff
@@ -451,24 +452,11 @@ def parse_metrics_from_report_log(report_log_line):
...
-            ),
-            metric_point_value,
-        )
-
-        initial_duration.add_tags(tags)
-
-        metrics.append(initial_duration)
-
     for metric_name in METRICS_TO_PARSE_FROM_REPORT:
+        # check if init duration is present
```
Comment (tianchu) [inline aws/logs_monitoring/enhanced_lambda_metrics.py]: Nits -> `# check whether the metric, e.g., init duration, is present in the REPORT log`

File: .github/workflows/lambdachecks.yml [hunk 3]
```diff
@@ -34,4 +34,4 @@ jobs:
           ./aws/logs_monitoring/trace_forwarder/scripts/run_tests.sh
       - name: Run enhanced metric unittest
         run: |
-          python -m unittest
+          python -m unittest discover ./aws/logs_monitoring/
```
Comment (tianchu) [inline .github/workflows/lambdachecks.yml:40]: Looks like you need to `pip install boto3` before running the tests.

Comment (jcstorms1) [reply .github/workflows/lambdachecks.yml:40]: Yep, added now

**Review by tianchu — APPROVED** [review]

**Pull request merged**